datastore/datastore.go

```diff
@@ -253,6 +254,29 @@ func NewDataStore(scope string, cfg *ScopeCfg) (DataStore, error) {
 	return newClient(scope, cfg.Client.Provider, cfg.Client.Address, cfg.Client.Config, cached)
 }
 
+// NewDataStoreFromConfig creates a new instance of LibKV data store starting from the datastore config data
+func NewDataStoreFromConfig(dsc discoverapi.DatastoreConfigData) (DataStore, error) {
+	sCfg, ok := dsc.Config.(*store.Config)
```

Comment:
This is not addressed yet. If `dsc.Config` is nil, then `ok` will be false. And that will cause the following check to fail.
I dont think that is the expected result. Can you please confirm ?
(We dont want to assume that the dsc.Config will never be nil in this function.)